backend/app/src/main/java/ch/zhaw/studyflow/webserver/http/query/QueryParametersImpl.java

```diff
@@ -0,0 +1,60 @@
+package ch.zhaw.studyflow.webserver.http.query;
+
+import java.util.*;
+
+public class QueryParametersImpl implements QueryParameters {
+    private final Map<String, List<String>> parameters;
+
+    public QueryParametersImpl(String query) {
+        this.parameters = parseQueryParameters(query);
+    }
+
+    @Override
+    public Set<String> keys() {
+        return parameters.keySet();
+    }
+
+    @Override
+    public Optional<String> getSingleValue(String key) {
+        if (key == null) {
+            throw new IllegalArgumentException("Key cannot be null");
```

Comment:
I would prefer to use ObjectsRequireNonNull here.
So it would be:
`Objects.requireNonNull(key, "Key cannot be null");`